# Formulas using LaTex

In [1]:
#import the libraries
import re
import pandas as pd
import latex

In [2]:
df = pd.DataFrame({'$y_i$': [5,4,3,1,3,71]})
df

$y_i$
0      5
1      4
2      3
3      1
4      3
5     71

## Assignment:
Calculate the Standard Deviation and the MAD (Median Absolute Deviation) for the dataframe 'df'

## Solution 

## Standard Deviation 

The formula for this --> 
1. variance ($s^2$) = $\frac{\sum \left({y_i} - {\bar y}\right)^2}{n - 1}$
2. standard deviation ($s$) = $\root \of{s^2}$ = root of the variance

So first we need to calculate the mean of the df:
1. The formula for the mean = $\bar y$ = $\frac{\sum y_i} n$
2. the sum of the data $y_i$ = 87 and there are 6 values
3. So if we enter the data --> $\frac{87} {6}$ = 14.5

Right let's add a column to the data frame with the numbers of the deviation $y_i - 14.5$

In [3]:
value = []
for x in df['$y_i$']:
    value.append(x- 14.5)
df.insert(loc = 1, column = '$y_i-14.5$', value = value)
df
    

$y_i$  $y_i-14.5$
0      5        -9.5
1      4       -10.5
2      3       -11.5
3      1       -13.5
4      3       -11.5
5     71        56.5

In [4]:
#Okay, now let's add a column with the values of $(y_i - 14.5)^2$

value_2 = []
for x in df['$y_i-14.5$']:
    value_2.append(x * x)
df.insert(loc = 2, column = '$(y_i-14.5)^2$', value = value_2)
df

$y_i$  $y_i-14.5$  $(y_i-14.5)^2$
0      5        -9.5           90.25
1      4       -10.5          110.25
2      3       -11.5          132.25
3      1       -13.5          182.25
4      3       -11.5          132.25
5     71        56.5         3192.25

In [5]:
#now let's add a total row to the dataframe: 
dfnew = df
dfnew.loc['Total']= dfnew.sum()
dfnew

$y_i$  $y_i-14.5$  $(y_i-14.5)^2$
0        5.0        -9.5           90.25
1        4.0       -10.5          110.25
2        3.0       -11.5          132.25
3        1.0       -13.5          182.25
4        3.0       -11.5          132.25
5       71.0        56.5         3192.25
Total   87.0         0.0         3839.50

Now let's use this data to fill in the formula and get the Standard Deviation:

1. variance ($s^2$) = $\frac{\sum \left({y_i} - {\bar y}\right)^2}{n - 1}$
2. The top part of the fraction is 3839.50
3. The lower part of the fraction is 6 - 1 = 5
4. Therefore the variance is $\frac {3839.50} {5}$ = 767.9
5. the Standard Deviation = $\root \of {767.9}$ = 27.711  

## MAD (Median Absolute Deviation) 

The formula for this --> MAD = median {|$y_1 - {\bar y}$|, |$y_2 - {\bar y}$|, ... , |$y_n - {\bar y}$|} / .6745

First we need to calculate the median of this df

1. The middle values are 3 + 4 (when put in ascending order) as there are 6 values.
2. The formula for the median therefore = $\tilde y$ = $\frac{3 + 4} {2}$ = 3.5


Okay, now let's make the dataframe again with a column $y_i - 3.5$ 

In [6]:
df = pd.DataFrame({'$y_i$': [5,4,3,1,3,71]})
value = []
for x in df['$y_i$']:
    value.append(x - 3.5)
df.insert(loc = 1, column = '$y_i - 3.5$', value = value)
df

$y_i$  $y_i - 3.5$
0      5          1.5
1      4          0.5
2      3         -0.5
3      1         -2.5
4      3         -0.5
5     71         67.5

In [7]:
#Okay, to make the calculation of the median absolute deviation correctly, 
#we have to make all values positive / absolute, so let's add another column. 
df['|$y_i - 3.5$|'] = df['$y_i - 3.5$'].abs()
df


$y_i$  $y_i - 3.5$  |$y_i - 3.5$|
0      5          1.5            1.5
1      4          0.5            0.5
2      3         -0.5            0.5
3      1         -2.5            2.5
4      3         -0.5            0.5
5     71         67.5           67.5

In [8]:
#right, now let's add a total row to this dataframe:
dfnew = df
dfnew.loc['Total']= dfnew.sum()
dfnew

$y_i$  $y_i - 3.5$  |$y_i - 3.5$|
0        5.0          1.5            1.5
1        4.0          0.5            0.5
2        3.0         -0.5            0.5
3        1.0         -2.5            2.5
4        3.0         -0.5            0.5
5       71.0         67.5           67.5
Total   87.0         66.0           73.0

Okay, let's fill in the formula:
1. first compute the median of the last column when placed in ascending order: $\frac{(0.5+1.5)} {2}$ = 1
2. Next, devide this median by .675, so MAD = 1.48
